### PR TITLE
[RPD-306] Add core stack add command

### PR DESCRIPTION
**Pull request by Christopher-Norman**
This PR adds the `matcha stack add <module type> <module flavor>` command. 

Upon a successful command run the `matcha.config.json` file will be updated:

```
 % matcha stack add experiment_tracker mlflow
Matcha 'experiment_tracker' module of flavor 'mlflow' has been added to the current stack.
```

`matcha.config.json` file:

```
{"stack": {"name": "custom", "experiment_tracker": "mlflow"}}
```
along with the stack name "custom"

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [x] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [x] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Other (add details above)

### CODE REVIEW

File: src/matcha_ml/core/core.py [hunk 1]
```diff
@@ -368,17 +368,53 @@ def stack_set(stack_name: str) -> None:
     MatchaConfigService.update(stack)
 
 
-def stack_add(module: str) -> None:
+STACK_MODULES = {
```
Comment (Christopher-Norman) [inline src/matcha_ml/core/core.py]: Move this

File: src/matcha_ml/cli/cli.py [hunk 2]
```diff
@@ -268,29 +268,39 @@ def set(stack: str = typer.Argument("default")) -> None:
 
 
 @stack_app.command(help="Add a module to the stack.")
-def add(module: str = typer.Argument(None)) -> None:
+def add(module: str = typer.Argument(None), flavor: str = typer.Argument(None)) -> None:
```
Comment (JonoCX) [inline src/matcha_ml/cli/cli.py]: If we have the following function signature:

```python
def add(module: Annotated[str, typer.Argument(None)], flavor: Annotated[str, typer.Argument(None)]) -> None:
```

Then we don't need to have the `if ... else ...` error handling in this function, Typer will take care of it for us and include the styling we want:

Screenshot 2023-08-29 at 12 56 11

File: src/matcha_ml/core/core.py [hunk 3]
```diff
@@ -368,17 +369,42 @@ def stack_set(stack_name: str) -> None:
...
     Raises:
         MatchaInputError: if the stack_name is not a valid stack type
         MatchaError: if there are already resources provisioned.
     """
-    ...
+    module_type = module_type.lower()
+    module_flavor = module_flavor.lower()
+
+    if RemoteStateManager().is_state_provisioned():
+        raise MatchaError(
```
Comment (JonoCX) [inline src/matcha_ml/core/core.py:387]: Don't forget to add tests that cover the exceptions in this function

**Review by JonoCX — CHANGES_REQUESTED** [review]
Nice one - see comments 🦖

**Review by swells2020 — APPROVED** [review]
LGTM; Jon's already picked up the flavor typer argument.

File: src/matcha_ml/cli/cli.py (hunk 2 above)
Comment (Christopher-Norman) [reply src/matcha_ml/cli/cli.py]: Updated

File: src/matcha_ml/core/core.py (hunk 3 above)
Comment (Christopher-Norman) [reply src/matcha_ml/core/core.py:387]: Added

**Review by JonoCX — APPROVED** [review]

**Review by JonoCX — APPROVED** [review]

**Pull request merged**